### DISCUSSION TITLE
Renovate runner security and outside traffic

### DISCUSSION THREAD

**Discussion by mikkotimoharju** [renovatebot/renovate#32497] [Request Help]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Gitlab renovate runner

### Please tell us more about your question or problem

We are using [Renovate runner](https://gitlab.com/renovate-bot/renovate-runner) for our private Gitlab.com projects. However, we are a bit concerned about the security of giving renovate access to the source code. Is there any way of making sure that the renovate runner as an open source project does not contain any malicious code that would send confidential data to 3rd party operators? How is the security handled and reviewed for commits in the renovatebot and renovate runner codebases?

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment]
All code in the GitLab runner is open source, including the code used to build images. The Renovate project executes continuous security tests in it's build systems to catch errors and vulnerabilities as early as possible.
